app/src/main/java/org/glucosio/android/activity/MainActivity.java

```diff
@@ -528,6 +533,99 @@ public void onClick(View v) {
+        exportDialogDateTo = (TextView) exportDialog.findViewById(R.id.activity_export_date_to);
+        exportRangeButton = (RadioButton) exportDialog.findViewById(R.id.activity_export_range);
+        final RadioButton exportAllButton = (RadioButton) exportDialog.findViewById(R.id.activity_export_all);
+        final TextView exportButton = (TextView) exportDialog.findViewById(R.id.dialog_export_add);
+        final TextView cancelButton = (TextView) exportDialog.findViewById(R.id.dialog_export_cancel);
+
+        exportRangeButton.setChecked(true);
+
+        exportDialogDateFrom.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Calendar now = Calendar.getInstance();
+                DatePickerDialog dpd = DatePickerDialog.newInstance(
+                        MainActivity.this,
+                        now.get(Calendar.YEAR),
+                        now.get(Calendar.MONTH),
+                        now.get(Calendar.DAY_OF_MONTH)
+                );
+                dpd.show(getFragmentManager(), "PDFfromDateDialog");
+                dpd.setMaxDate(now);
+            }
+        });
+
+        exportDialogDateTo.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Calendar now = Calendar.getInstance();
+                DatePickerDialog dpd = DatePickerDialog.newInstance(
+                        MainActivity.this,
+                        now.get(Calendar.YEAR),
+                        now.get(Calendar.MONTH),
+                        now.get(Calendar.DAY_OF_MONTH)
+                );
+                dpd.show(getFragmentManager(), "PDFtoDateDialog");
+                dpd.setMaxDate(now);
+            }
+        });
+
+        exportRangeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                boolean isChecked = exportRangeButton.isChecked();
+                exportDialogDateFrom.setEnabled(true);
+                exportDialogDateTo.setEnabled(true);
+                exportAllButton.setChecked(!isChecked);
+            }
+        });
+
+        exportAllButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                boolean isChecked = exportAllButton.isChecked();
+                exportDialogDateFrom.setEnabled(false);
+                exportDialogDateTo.setEnabled(false);
+                exportRangeButton.setChecked(!isChecked);
+                exportButton.setEnabled(true);
+            }
+        });
+
+        exportButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (validateExportDialog()) {
+                    exportPdfPresenter.onExportClicked(exportAllButton.isChecked());
```

Comment:
This point is only one difference from another export dialog. Why not to generalise and reuse it?